job-runner/webapp/server/classes/UpdateCbioData.js

```diff
@@ -0,0 +1,155 @@
+function UpdateCbioData(job_id) {
+  console.log('run UpdateCbioData', this, 'job', job_id)
+  Job.call(this, job_id);
+  // UpdateCbioData
+  // args
+  //  sample_group_id - samples with expression and clinical data for upload into cbio
+}
+UpdateCbioData.prototype = Object.create(Job.prototype);
+UpdateCbioData.prototype.constructor = UpdateCbioData;
+
+UpdateCbioData.prototype.run = function () {
+  // create paths for files on the disk
+  // Use temporary folders at /tmp/UpdateCbioData[job_id]
+
+  var workDir = "/tmp/" + "UpdateCbioData_" + this.job._id;
+  console.log('workdir',workDir)
+
+  try {
+    fs.mkdirSync(workDir);
+  } catch (e) {
+    console.log("Pretty sure you reran the job: {$set: { status: 'waiting' }}");
+    console.log("error:", e);
+    throw e;
+  }
+
+  console.log("workDir: ", workDir);
+
+  // create a sample group which is the combination of the two sample groups
+  // so that we can easily write out a file
+
+  var group = SampleGroups.findOne(this.job.args.sample_group_id);
+
+  // combine samples of same data set into single array
+  var dataSetHash = {};
+  var dataSetName = "";
+  _.each(group.data_sets, function (dataSet) {
+    // check if we've seen this data set already
+    var seenAlready = dataSetHash[dataSet.data_set_id];
+    if (!seenAlready) {
+      // if we haven't, set it up
+      dataSetName = dataSet.data_set_name;
+      seenAlready = {
+        data_set_name: dataSet.data_set_name,
+        sample_labels: [],
+      };
+    }
+
+    // combine the samples together
+    seenAlready.sample_labels =
+        seenAlready.sample_labels.concat(dataSet.sample_labels)
+    dataSetHash[dataSet.data_set_id] = seenAlready;
+  });
+  var comboSampleGroupDataSets = _.map(dataSetHash,
+      function (samplesAndName, data_set_id) {
+    return {
+      data_set_id: data_set_id,
+      data_set_name: samplesAndName.data_set_name,
+      sample_labels: samplesAndName.sample_labels,
+
+      // I think we can fake this
+      unfiltered_sample_count: 1,
+    };
+  });
+
+  //var comboSampleGroupId = SampleGroups.insert({
+  //  name: "temp - created in UpdateCbioData to call an adapter",
+  //  version: 1,
+  //  data_sets: comboSampleGroupDataSets,
+  //  value_type: group.value_type,
+
+    // invisible
+  //  collaborations: [],
+  //});
+
+  // star the promise chain: woohoo!
+
+  var self = this;
+  var deferred = Q.defer();
+
+  // define up here so as to be available throughout promise chain (so that
+  // we can skip a .then block)
+  var geneSetGroupPath;
+  var expressionDataPath = path.join(workDir, "data_expression.txt");
+  var logPath = path.join(workDir, "*.log");
+  var associated_object = {
+    collection_name: "SampleGroups",
+    mongo_id: group._id,
+  };
+  var form_id = this.job.args.form_id;
```

Comment:
Ah, the `form_id` is indeed used below. `geneSetGroupPath` isn't, however (a couple lines up).